docs/learn/learn-xcm.md

```diff
@@ -0,0 +1,299 @@
+---
+id: learn-xcm
+title: Cross-Consensus Message Format (XCM)
+sidebar_label: Cross-Consensus Message Format (XCM)
+description: Learn about the messaging format at the forefront of interoperability.
+keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
+slug: ../learn-xcm
+---
+
+Cross-Consensus Message Format(XCM) aims to be a language to communicate ideas between consensus systems. One
+of Polkadot's promises is that of interoperability, and XCM is the vehicle by which it will deliver this
+promise. Simply, it is a standard that allows protocol developers to define the data and origins which
+their chains can send and recieve from. Out of the box it comes with a VM that allows for customization
+of execution as well as the following properties:
+
+1. **Asynchronous**: XCM messages in no way assume that the sender will be blocking on its
+   completion.
+2. **Absolute**: XCM messages are guaranteed to be delivered and interpreted accurately, in order
+   and in a timely fashion.
+3. **Asymmetric**: XCM messages do not have results. Any results must be separately communicated to
+   the sender with an additional message.
+4. **Agnostic**: XCM makes no assumptions about the nature of the Consensus System between which
+   messages are being passed.
+
+:::note 
+
+XCM is a work-in-progress. XCM v2 is deployed on Polkadot and v3 is currently in
+development. See Gavin's talk about v3 in the resources section.
+
+:::
+
+## A Format, Not a Protocol
+
+What started as an approach to _cross-chain communication_, has evolved into a format for
+[**Cross-Consensus Communication**](https://polkadot.network/cross-chain-communication) that is
+not only conducted between chains, but also between smart contracts, pallets, bridges, and even
+sharded enclaves like [SPREE](learn-spree.md).
+
+**XCM is related to cross-chain in the same way that REST is related to RESTful.**
```

Comment:
Wondering how this analogy works. RESTful A web API that obeys the REST constraints. I'd go with the analogy below where XCMP is defined.

**XCM is related to XCMP in the same way that REST is related to RESTful.**

Reply:
Good point, moving this there.